### PR TITLE
#include hyperlink to included file

### PR DESCRIPTION
**Pull request by antoni-rokitnicki**
https://issues.jboss.org/browse/JBIDE-19966

freemarkerIDE
#include hyperlink to included file

### CODE REVIEW

File: plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	public IncludeHyperlinkDetector(ITextViewer textViewer, Editor editor) {
+		this.editor = editor;
+	}
+
+	@Override
+	public IHyperlink[] detectHyperlinks(ITextViewer textViewer, IRegion region, boolean canShowMultipleHyperlinks) {
+		Item item = editor.getItemSet().getItem(region.getOffset());
+		if (item instanceof GenericDirective) {
+			GenericDirective genericDirective = (GenericDirective) item;
+			if ("include".equals(genericDirective.getName())) { //$NON-NLS-1$
```
Comment (dgolovin) [inline plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java]: Should be:

``` java
Keyword.INCLUDE.toString().equals(genericDirective.getName())
```

File: plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	private Editor editor;
+	public IncludeHyperlinkDetector(ITextViewer textViewer, Editor editor) {
+		this.editor = editor;
+	}
+
+	@Override
+	public IHyperlink[] detectHyperlinks(ITextViewer textViewer, IRegion region, boolean canShowMultipleHyperlinks) {
+		Item item = editor.getItemSet().getItem(region.getOffset());
+		if (item instanceof GenericDirective) {
```
Comment (dgolovin) [inline plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java]: GenericDirective could be replaced to AbstractItem to use more generic abstraction.

**Comment by dgolovin** [comment]
Thank yo for PR!

Do you plan to have any junit tests to cover new functionality?

File: plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			GenericDirective genericDirective = (GenericDirective) item;
+			if ("include".equals(genericDirective.getName())) { //$NON-NLS-1$
+				String[] splitContents = genericDirective.splitContents();
+				if (splitContents.length>1) {
+					String text = splitContents[1].replace("\"", "");  //$NON-NLS-1$//$NON-NLS-2$
+					IFile file = genericDirective.getResource().getParent().getFile(new Path(text));
```
Comment (fbricon) [inline plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java]: will it support relative paths like ../../foo/bar?

Comment (dgolovin) [inline plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/IncludeHyperlinkDetector.java]: Tested locally and it doesn't support relative paths.

**Comment by dgolovin** [comment]
this pr was released to master.

**Pull request closed without merging**